server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
-  const users = await User.findAll({ attributes: User.allowedFields });
+  const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
+  const limit = 20;
+  const offset = (page - 1) * limit;
 
-  res.json(users);
+  if (!Number.isInteger(page) || page < 1) {
+    return res.status(400).json({ error: 'Invalid page number' });
+  }
+
+  // Fine all users, using the limit and offset to implement offset pagination
+  const users = await User.findAll({
+    attributes: User.allowedFields,
+    limit: limit,
+    offset: offset,
+  });
+
+  const count = await User.count();
```

Comment:
these two can (and probably should) happen concurrently via `Promise.all`

Reply:
I'll agree and I'll raise you a `findAndCountAll()` ([Docs](https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findandcountall))